app/src/main/cpp/skyline/vfs/directory.h

```diff
@@ -19,17 +22,19 @@ namespace skyline::vfs {
         struct Entry {
             std::string name;
             EntryType type;
+            size_t size; //!< 0 if a directory
         };
 
         /**
          * @brief A descriptor for what will be returned when reading a directories contents
+         * @note MUST match HOS
          */
         union ListMode {
             struct {
                 bool directory : 1; //!< The directory listing will contain subdirectories
                 bool file : 1; //!< The directory listing will contain files
             };
-            u32 raw;
+            u32 raw{};
```

Comment:
This doesn't affect anything unless you move it to above the `struct`

Reply:
nope, it does, it's just that unions only zero the first entry by default with {}. Adding {} makes it also zero out that as we discussed ages back. I did it this way as it allows you to use ListMode{} neatly.

Reply:
You're wrong, the entire issue was that regardless of what you do it'll only init the first member in unions. You will need to change the order if you want the entire u32 to be 0-init'd.

Reply:
You sure, I don't remember discussing that at all?